### PR TITLE
feat: add invariants to x/foundation

### PR DESCRIPTION
**Pull request by 0Tech**
## Description

This patch will add following invariants to x/foundation:
- ModuleAccountInvariant
  - It checks whether each balances of the module accounts is in a sane state.
    - "treasury": It must be equal to the corresponding value in the pool.
- TotalWeightInvariant
  - It checks whether the total weight is in a sane state.
    - Each weight of the member is implicitly one, hence the weight must be equal to the number of the members.
- ProposalInvariant
  - It checks whether each proposal is in a sane state.
    - Old proposals must have been aborted, so they must not be in the submitted state.

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Review by zemyblue — COMMENTED** [review]
If the invariants is broken, how can we resolve in the chain?

**Comment by 0Tech** [comment]
> If the invariants is broken, how can we resolve in the chain?

Halting the chain is the current short-term solution. The goal of the invariant feature is to prevent ripple effects from the violated invariant. Fixing the cause is not the goal of this feature.

File: x/foundation/client/testutil/grpc.go [hunk 1]
```diff
@@ -31,7 +30,6 @@ func (s *IntegrationTestSuite) TestGRPCParams() {
 				Params: foundation.Params{
 					FoundationTax: sdk.MustNewDecFromStr("0.2"),
 					CensoredMsgTypeUrls: []string{
-						sdk.MsgTypeURL((*stakingtypes.MsgCreateValidator)(nil)),
```
Comment (zemyblue) [inline x/foundation/client/testutil/grpc.go:34]: Why does this checking remove?

Comment (0Tech) [reply x/foundation/client/testutil/grpc.go:34]: Because, the suite has removed it from the state.

**Review by zemyblue — APPROVED** [review]

**Review by dudong2 — APPROVED** [review]

**Pull request merged**